pkg/frontend/transport/handler.go

```diff
@@ -190,6 +192,20 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	activityIndex := f.at.Insert(func() string { return httpRequestActivity(r, r.Header.Get("User-Agent"), params) })
 	defer f.at.Delete(activityIndex)
 
+	if isStreamingEndpoint(r) {
+		// Set write deadline for streaming responses.
+		var deadline time.Time
+		if f.cfg.ResponseStreamTimeout > 0 {
+			deadline = time.Now().Add(f.cfg.ResponseStreamTimeout)
+		}
+		err = http.NewResponseController(w).SetWriteDeadline(deadline)
```

Comment:
Can you confirm we don't need to also set the read deadline, right?

Reply:
Correct, the requests are typically small and the default read deadline should be applied.